pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -808,7 +811,8 @@ private CompletableFuture<Consumer> internalSubscribe(final TransportCnx cnx, St
             CompletableFuture<Consumer> future = subscriptionFuture.thenCompose(subscription -> {
                 Consumer consumer = new Consumer(subscription, subType, topic, consumerId, priorityLevel,
                         consumerName, isDurable, cnx, cnx.getAuthRole(), metadata,
-                        readCompacted, keySharedMeta, startMessageId, consumerEpoch);
+                        readCompacted, keySharedMeta, startMessageId, consumerEpoch,
+                        schemaData == null ? SchemaType.AUTO_CONSUME : schemaData.getType());
```

Comment:
Null `schemaData` can also represent BYTES schema. After this change, it will not only fix the auto_consume schema issue. It also changed the behavior of the BYTES schema.

Before, if there were consumers with BYTES schema, only the consumer with BYTES schema could subscribe to the topic. With this fix, the consumer with Avro schema can also subscribe to this topic, right?

Looks like we are not able to fix the problem without client-side changes.

Reply:
it seems so, if one consumer with BYTES schema sub the topic first, the second consumer with AVRO schema can also sub the topic. If the broker can't distinguish the BYTES schema and AUTO_CONSUME, this problem can't be solved. It can only be distinguished by sub with the Schema type from the client, there is no other way